app/oauth.go

```diff
@@ -501,6 +501,10 @@ func (a *App) DeauthorizeOAuthAppForUser(userID, appID string) *model.AppError {
 		}
 	}
 
+	if err := a.Srv().Store().OAuth().RemoveAuthDataByClientId(appID, userID); err != nil {
```

Comment:
Given this is a bugfix, I think it would make sense to have a test case for `DeauthorizeOAuthAppForUser` that reveals the issue and that the fix leads to the expected behaviour (I see the method is not tested in `oauth_test`)